pyccel/codegen/printing/ccode.py

```diff
@@ -1387,6 +1387,9 @@ def _print_Declare(self, expr):
             assert init == ''
             preface = ''
             init    = ' = {.shape = NULL}'
+        elif declaration_type.startswith('vec'):
```

Comment:
Safer to check the type exactly

```suggestion
        elif isinstance(var.class_type, HomogeneousListType):
```